ocaml/typing/parmatch.ml

```diff
@@ -1799,6 +1819,13 @@ and record_lubs l1 l2 =
         (lid1, lbl1,lub p1 p2)::lub_rec rem1 rem2 in
   lub_rec l1 l2
 
+and tuple_lubs ps qs = match ps,qs with
+| [], [] -> []
+| (p_label, p)::ps, (q_label, q)::qs
+      when Option.equal String.equal p_label q_label ->
+    (p_label, lub p q) :: tuple_lubs ps qs
+| _,_ -> raise Empty
```

Comment:
This seems correct but it's not what the original `lubs` does, which returns `[]` rather than raising on different-length tuples. The old code seems wrong? (Probably it's impossible to hit, since this only runs on patterns that have passed typechecking at the same type?)

Reply:
I had the same question when I read this and came to the same conclusion: This isn't reachable because `lub` is only used after typechecking, but if it were the new behavior seems more correct.

Reply:
I haven’t looked at the code in question but from these comments I’d assume it was hit in the case of GADTs. Most of the code in parmatch needs to tolerate patterns that don’t agree being in the same column due to GADTs.